src/Foundatio/Caching/InMemoryCacheClient.cs

```diff
@@ -401,7 +401,10 @@ public InMemoryCacheClient(Builder<InMemoryCacheClientOptionsBuilder, InMemoryCa
 
                     collection.Add(stringValue);
                     cacheEntry.Value = collection;
-                    cacheEntry.ExpiresAt = expiresAt;
+
+                    if (expiresIn.HasValue)
```

Comment:
Because if you pass in null for the expiresIn argument, we shouldn't update any existing expiresAt (local var computed to DateTime.MaxValue in a tenary argument).

Reply:
That makes sense for updating an existing entry, but if it's a new entry I think we need to set it to the default value.

Reply:
It's correct:

```cs

                var entry = new CacheEntry(items, expiresAt, _shouldClone);
                _memory.AddOrUpdate(key, entry, (k, cacheEntry) => {
```
If it's an add the cache entry will use the expiresAt, if it's an update it will selectively update it.